### PR TITLE
#162171029 Add pagination support to articles

### PR DESCRIPTION
**Pull request by pitaz**
#**What does this PR do?**
- Add pagination support to articles

**Description of Task to be completed?**
- create ArticlesController.js

**How should this be manually tested?**
- clone the repository `https://github.com/andela/heimdal-ah-backend.git`
- run `npm run install`
- run `npm run start:dev`
- input `api/v1/articles` on postman with the `get` method to fetch all articles

**Any background context you want to provide?**
- none

**What are the relevant pivotal tracker stories?**
- [#162171029](https://www.pivotaltracker.com/story/show/162171029)

**Screenshots (if appropriate)**
- NA
**Questions:**
- NA

### CODE REVIEW

File: middlewares/articleMiddleware.js [hunk 1]
```diff
@@ -20,25 +20,36 @@ const checkArticle = async (req, res, next) => {
...
-    const fetchArticle = await articles.findOne({
-      where: {
-        id: req.params.id
-      }
-    });
-    if (!fetchArticle) {
-      return StatusResponse.notfound(res, {
-        message: 'Could not find article'
-      });
+// const findArticleById = async (req, res, next) => {
```
Comment (azupatrick0) [inline middlewares/articleMiddleware.js]: @pitaz, if line 23 to line39 is not in use, is it not better we remove them?

Comment (pitaz) [reply middlewares/articleMiddleware.js]: Thanks @azupatrick0 This has been resolved and will reflect on next PR.

File: helpers/pagination.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const pagination = (page, size, count) => {
+  let currentPage = Math.abs(Number(page)) || 1;
+  let offset = 0;
+  let limit = 2;
```
Comment (azupatrick0) [inline helpers/pagination.js]: hello @pitaz nice work so far, but from the API spec, the limit should be 20 articles.

File: middlewares/articleMiddleware.js (hunk 1 above)
Comment (azupatrick0) [reply middlewares/articleMiddleware.js]: Kudos @pitaz

File: helpers/pagination.js (hunk 2 above)
Comment (pitaz) [reply helpers/pagination.js]: Thanks @azupatrick0 This has been resolved.

Comment (azupatrick0) [reply helpers/pagination.js]: Kudos @pitaz

**Pull request merged**